spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/GatewayMetricsFilter.java

```diff
@@ -24,26 +24,27 @@
 import org.apache.commons.logging.LogFactory;
 import reactor.core.publisher.Mono;
 
-import org.springframework.cloud.gateway.route.Route;
+import org.springframework.cloud.gateway.support.tagsprovider.GatewayTagsProvider;
 import org.springframework.core.Ordered;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.server.reactive.AbstractServerHttpResponse;
 import org.springframework.http.server.reactive.ServerHttpResponse;
 import org.springframework.web.server.ServerWebExchange;
 
-import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
-
 /**
  * @author Tony Clarke
+ * @author Ingyu Hwang
  */
 public class GatewayMetricsFilter implements GlobalFilter, Ordered {
 
 	private static final Log log = LogFactory.getLog(GatewayMetricsFilter.class);
 
 	private MeterRegistry meterRegistry;
 
-	public GatewayMetricsFilter(MeterRegistry meterRegistry) {
```

Comment:
Since this is a public class, please put the original back and deprecate it and pass the default impl to the new constructor.

Reply:
Fixed